node/packages/parser/src/ethereum/index.ts

```diff
@@ -59,7 +60,10 @@ export class TransactionParser {
     const receiveAddress = tx.vout[0].addresses?.[0] ?? ''
 
     // We expect only one Parser to return a result. If multiple do, we take the first and early exit.
-    const result = await findAsyncSequential(this.parsers, async (parser) => await parser.parse(tx))
+    const contractParserResult: TxSpecific<ParseTx> | undefined = await findAsyncSequential(
```

Comment:
If we update the return type of the predicate parameter in `findAsyncSequential` to `Promise<U | undefined>` to match the signature of contract `parse()`, we can then utilize the generic types and get the correct return type without needing to cast `findAsyncSequential<Parser, TxSpecific<ParseTx>>()`

Reply:
Very clean. Fixed in https://github.com/shapeshift/unchained/pull/247/commits/0b9cbee47744a18398c252dc10bcda1a33989f92.